drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -679,6 +700,82 @@ def has_submission_deadline_passed(self):
         else:
             return False
 
+    @property
+    def closed(self):
+        """
+        Is the student still allowed to submit answers?
+        """
+        if self.used_all_attempts:
+            return True
+        if self.has_submission_deadline_passed:
+            return True
+
+        return False
+
+    @property
+    def is_attempted(self):
+        """
+        Has the problem been attempted?
+        """
+        return self.attempts > 0
+
+    @property
+    def is_finished(self):
+        """
+        Returns True if answer is closed or answer is correct.
+        """
+        return self.closed or self.is_correct
+
+    @property
+    def used_all_attempts(self):
+        """
+        Returns True if student has used all allowed attempts.
+        """
+        if self.max_attempts is None or self.max_attempts == 0:
+            return False
+        return self.attempts >= self.max_attempts
+
+    @property
+    def is_correct(self):
+        """
+        Returns True if answer is correct.
+        """
+        return self._is_answer_correct()
```

Comment:
We already have `self._is_answer_correct()`. Why don't we use it directly in `is_answer_available` instead of introducing a new layer of methods? If we want to to call it `is_correct`, then we should rename `_is_answer_correct` instead of introducing a chain of one-line methods.